src/api/getRouteCSM.ts

```diff
@@ -0,0 +1,88 @@
+// Adjacent waypoints that are closer than 4 metres when resolved onto the cycle network are treated as duplicates and ignored.
+
+//new type for itinerary points
+export type ItineraryPoint = {
+    lat: number;
+    lon: number;
+    name: string; //can 
+};
+
+// plan string
+// The type of route, which can be one of the values: balanced, fastest, quietest. There is also shortest but see the notes below.
+// balanced: We recommend this to be the default route type in your interface - it aims to give practical routes that balance speed and pleasantness, suitable for most riders.
+// fastest: This route type will tend to favour busier roads that suit more confident riders.
+// quietest: The route type will produce more pleasant, but often less direct, routes.
+// shortest: In general we do not recommend including this in your interface unless you have a need for it, as this will not give particularly practical routes. These will be literally the shortest route, with only land ownership rights causing any deviation from this. It will suggest, for instance, dismounting and walking down the opposite direction of a one-way street, and will gladly route over the top of a hill when that is the shortest distance.
+
+export enum planType {
+    balanced = "balanced",
+    fastest = "fastest",
+    quietest = "quietest",
+    shortest = "shortest"
+}
+
+//function to convert itinerary points to string
+const itineraryPointsToString = (itineraryPoints: ItineraryPoint[]) => {
+    let itineraryPointsString = "";
+    //loop through itinerary points and add to string if name is empty omit it
+    itineraryPoints.forEach((point) => {
+        if (point.name === "") {
+            itineraryPointsString += `${point.lon},${point.lat}|`;
+        } else {
+            //url encode the name
+            const name = encodeURIComponent(point.name);
+            itineraryPointsString += `${point.lon},${point.lat},${name}|`;
+        }
+    });
+    return itineraryPointsString.slice(0, -1);
+}
+/**
+ * 
+ * @param itineraryPoints 
+ * @param plan 
+ * @returns Return values, in detail https://www.cyclestreets.net/api/v1/journey/#jpReturn
+ * 
+ */
+export const fetchRouteCSM = async (itineraryPoints: ItineraryPoint[], plan: planType) => {
+    const url = `https://www.cyclestreets.net/api/journey.json?key=${ConfigEnv.apiKey}&reporterrors=1&itinerarypoints=${itineraryPointsToString(itineraryPoints)}&plan=${plan}`;
+    const response = await fetch(url);
+    const data = await response.json();
+    return data;
+}
+
+export const testRouteCSM = async () => {
```

Comment:
dodałabym do tych funkcji już parametry na przyszłość a mocki wyrzuciła na zewnątrz funkcji

Reply:
to funkcja obrazująca działanie fetchRouteCSM.